File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i ++){
        
        map.put(nums[i], i);     
    } 
    for(int i = 0; i < nums.length; i ++){
        int compliment = target - nums[i];
        if(map.containsKey(compliment) && map.get(compliment) != i){
            int[] indices = {i, map.get(compliment)};
            return indices;
        }
        
    }
    return null;
}
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        i = int(len(nums)/2)-1
        mx = len(nums)-1
        mn=0
        while(mn <= mx):
            i=int((mx-mn)/2 + mn)
            if(nums[i] == target):
                return i
            elif(nums[i] > target):
                mx-=1
            else:
                mn+=1
                
        return -1